config/routes.rb

```diff
@@ -7,4 +7,8 @@
 
   # Defines the root path route ("/")
   # root "posts#index"
+  get '/users', to: 'users#index'
+  get '/users/:id', to: 'users#show', as: 'user'
+  get '/users/:user_id/posts', to: 'posts#index'
+  get '/posts/:id', to: 'posts#show'
```

Comment:
- [ ] The post show route should be nested within the users route.

  ```
  get '/users/:user_id/posts/:id', to: 'posts#show'
  ```

- [ ] **[OPTIONAL]** Alternatively, all these routes can be defined more concisely.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```